pkg/kubectl/cmd/util/helpers.go

```diff
@@ -400,22 +404,70 @@ func DumpReaderToFile(reader io.Reader, filename string) error {
 	return nil
 }
 
+// UpdateApplyAnnotation serializes the VersionedObject of an info, puts it into
+// an annotation on the Object that can be used by RunApply, and returns both the
+// original and modified contents of the annotation.
+func UpdateApplyAnnotation(info *resource.Info) ([]byte, []byte, error) {
+	// Get the current annotations from the object.
+	annotations, err := info.Mapping.MetadataAccessor.Annotations(info.Object)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// If there weren't any, then make a new map of annotations.
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	// Retrieve the original configuration from the annotations, if present.
+	original := []byte(annotations[ApplyAnnotation])
+
+	var modified []byte
+	if info.VersionedObject != nil {
```

Comment:
Should this reset the apply annotation on the versioned object before serializing it?  It seems like this would add the annotation every edit recursively, eventually leading to exceeding the max annotation size.